flake.nix

```
@@ -12,16 +12,15 @@
       nixpkgs,
       utils,
     }:
-    let
-      pkgsForSystem =
-        system:
-        import nixpkgs {
-          inherit system;
-        };
-    in
     utils.lib.eachDefaultSystem (system: rec {
-      legacyPackages = pkgsForSystem system;
-      devShells.default = legacyPackages.callPackage ./nix/dev-shell.nix { };
+      legacyPackages = nixpkgs.legacyPackages.${system}.extend (
+        final: prev: {
+          dev-server = prev.callPackage ./nix/dev-server.nix { };
+          dev-shell = prev.callPackage ./nix/dev-shell.nix { };
```

```suggestion
          dev-server = final.callPackage ./nix/dev-server.nix { };
          dev-shell = final.callPackage ./nix/dev-shell.nix { };
```

generally you use final, so you can pass sibling packages between each other.